File: app/src/main/java/com/mmutert/freshfreezer/data/converters/UUIDConverter.java
package com.mmutert.freshfreezer.data.converters;

import androidx.room.TypeConverter;

import java.util.Date;
import java.util.UUID;


public class UUIDConverter {

    @TypeConverter
    public static UUID toUUID(String uuid){
        return uuid == null ? null: UUID.fromString(uuid);
    }

    @TypeConverter
    public static String fromUUID(UUID uuid){
        return uuid == null ? null : uuid.toString();
    }
}

File: app/src/main/java/com/mmutert/freshfreezer/data/converters/UUIDConverter.kt
package com.mmutert.freshfreezer.data.converters

import androidx.room.TypeConverter
import java.util.*

object UUIDConverter {
    @JvmStatic
    @TypeConverter
    fun toUUID(uuid: String?): UUID? {
        return if (uuid == null) null else UUID.fromString(uuid)
    }

    @JvmStatic
    @TypeConverter
    fun fromUUID(uuid: UUID?): String? {
        return uuid?.toString()
    }
}